packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -604,23 +605,28 @@ class StripeHelper {
    * validate this is an appropraite change for tier use.
    *
    * @param {Subscription['id']} subscriptionId
-   * @param {Plan['id']} planId
+   * @param {Plan['id']} newPlanId
    */
-  async changeSubscriptionPlan(subscriptionId, planId) {
+  async changeSubscriptionPlan(subscriptionId, newPlanId) {
     const subscription = await this.stripe.subscriptions.retrieve(
       subscriptionId
     );
-    if (subscription.items.data[0].plan.id === planId) {
+    const currentPlanId = subscription.items.data[0].plan.id;
+    if (currentPlanId === newPlanId) {
       throw error.subscriptionAlreadyChanged();
     }
     return await this.stripe.subscriptions.update(subscriptionId, {
       cancel_at_period_end: false,
       items: [
         {
           id: subscription.items.data[0].id,
-          plan: planId,
+          plan: newPlanId,
         },
       ],
+      metadata: {
+        previous_plan_id: currentPlanId,
+        plan_change_date: moment().unix(),
```

Comment:
Do we need moment here? Would `Math.floor(Date.now() / 1000)` would give the same result?

Reply:
I thought that it would read more clearly to use `moment().unix()`. A new contributor may not realize that `Math.floor(Date.now() / 1000)` results in a Unix timestamp in seconds.

Reply:
You are also correct that it made testing easier, since I could easily stub the result 😅

Reply:
Cool, those sound like good reasons to me! :+1: